packages/vite/src/utils/generator-util.spec.ts

```diff
@@ -0,0 +1,206 @@
+      );
+    });
+  });
+
+  describe('findExistingTargetsInProject', () => {
+    it('should return the correct targets', () => {
+      mockReactAppGenerator(tree);
+      const { targets } = readProjectConfiguration(tree, 'my-test-react-app');
+
+      const existingTargets = findExistingTargetsInProject(targets);
+      expect(existingTargets).toMatchObject({
+        buildTarget: 'build',
+        serveTarget: 'serve',
+        testTarget: 'test',
+      });
+    });
+
+    it('should return the correct - undefined - targets for Angular apps', () => {
+      mockAngularAppGenerator(tree);
+      const { targets } = readProjectConfiguration(tree, 'my-test-angular-app');
+      const existingTargets = findExistingTargetsInProject(targets);
+      expect(existingTargets).toMatchObject({
+        buildTarget: undefined,
+        serveTarget: undefined,
+        testTarget: 'test',
+        unsupported: true,
+      });
+    });
+  });
+
+  describe('handleUnsupportedUserProvidedTargets', () => {
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+    it('should throw error if unsupported and user does not want to confirm', async () => {
+      const { Confirm } = require('enquirer');
+      const confirmSpy = jest.spyOn(Confirm.prototype, 'run');
+      confirmSpy.mockResolvedValueOnce(false);
+      expect.assertions(2);
+      const object = {
+        unsupportedUserProvidedTarget: {
+          build: true,
+        },
+        userProvidedTargets: {
+          build: 'my-build',
+          serve: 'my-serve',
+          test: 'my-test',
+        },
+        targets: {
+          build: 'build',
+          serve: 'serve',
+          test: 'test',
+        },
+      };
+
+      try {
+        await handleUnsupportedUserProvidedTargets(
+          object.unsupportedUserProvidedTarget,
+          object.userProvidedTargets,
+          object.targets
+        );
+      } catch (e) {
+        expect(e).toBeDefined();
+        expect(e.toString()).toContain(
+          'The build target my-build cannot be converted to use the @nrwl/vite:build executor'
+        );
+      }
+    });
+
+    it('should NOT throw error if unsupported and user confirms', async () => {
+      const { Confirm } = require('enquirer');
+      const confirmSpy = jest.spyOn(Confirm.prototype, 'run');
+      confirmSpy.mockResolvedValue(true);
+      expect.assertions(2);
+      const object = {
+        unsupportedUserProvidedTarget: {
+          build: true,
+        },
+        userProvidedTargets: {
+          build: 'my-build',
+          serve: 'my-serve',
+          test: 'my-test',
+        },
+        targets: {
+          build: 'build',
+          serve: 'serve',
+          test: 'test',
+        },
+      };
+
+      expect(
+        handleUnsupportedUserProvidedTargets(
+          object.unsupportedUserProvidedTarget,
+          object.userProvidedTargets,
+          object.targets
+        )
+      ).resolves.toBeUndefined();
+
+      try {
```

Comment:
I don't think we need this try-catch since the error will fail the test regardless.